### PR TITLE
Add ODataPathInfo tests

### PR DESCRIPTION
**Pull request by gathogojr**
### Description

Related to #2978. Add `ODataPathInfo` tests for .NET 8.0

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### CODE REVIEW

**Review by WanjohiSammy — COMMENTED** [review]
Could you also add a test for ODataPathInfo() that takes `IEdmType` and `IEdmNavigationSource` as arguments
```cs
public ODataPathInfo(IEdmType targetEdmType, IEdmNavigationSource targetNavigationSource)
```

**Review by WanjohiSammy — DISMISSED** [review]
Could you also add a test for ODataPathInfo() that takes `IEdmType` and `IEdmNavigationSource` as arguments
```cs
public ODataPathInfo(IEdmType targetEdmType, IEdmNavigationSource targetNavigationSource)
```

**Review by marabooy — APPROVED** [review]

**Comment by gathogojr** [comment]
> Could you also add a test for ODataPathInfo() that takes `IEdmType` and `IEdmNavigationSource` as arguments
> 
> ```cs
> public ODataPathInfo(IEdmType targetEdmType, IEdmNavigationSource targetNavigationSource)
> ```

@WanjohiSammy There's some background that I didn't reference in this pull request

There's this related pull request https://github.com/OData/odata.net/pull/2978

An external contributor contributed this other pull request https://github.com/OData/odata.net/pull/2974 to fix a bug in our code when the target framework is net8.0.
What you might notice about the pull request is that it had no tests and it was merged before we could ask the contributor to add.

My pull request was intended to add tests for the changes in the contributor's pull request.

Basically, out of the two `ODataPathInfo` constructors, we didn't have tests for one that accepts an `ODataPath` argument so a bug went uncaught. For the other constructor that accepts an `IEdmType` and an `IEdmNavigationSource` arguments - that is the subject of your comment, it's referenced extensively in existing tests.

**Review by WanjohiSammy — APPROVED** [review]

**Review by marabooy — APPROVED** [review]
